homeassistant/components/vegehub/switch.py

```diff
@@ -0,0 +1,163 @@
+"""Switch configuration for VegeHub integration."""
+
+from typing import Any
+
+from vegehub import VegeHub
+
+from homeassistant.components.switch import (
+    SwitchDeviceClass,
+    SwitchEntity,
+    SwitchEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfElectricPotential
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, MANUFACTURER, MODEL
+
+SWITCH_TYPE = SwitchEntityDescription(
+    key="switch",
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Vegetronix sensors from a config entry."""
+    sensors = []
+    num_sensors = int(config_entry.data.get("hub", {}).get("num_channels") or 0)
+    num_actuators = int(config_entry.data.get("hub", {}).get("num_actuators") or 0)
+    is_ac = int(config_entry.data.get("hub", {}).get("is_ac") or 0)
+
+    num_batteries = 1
+    if is_ac:
+        num_batteries = 0
+
+    # We add up the number of sensors, plus the number of actuators, then add one
+    # for battery reading, and one because the array is 1 based instead of 0 based.
+    for i in range(
+        num_sensors + 1, num_sensors + num_actuators + 1
+    ):  # Add 1 for battery
+        if i > num_sensors:
+            name = f"VegeHub Actuator {i - num_sensors}"
+            sensor = VegeHubSwitch(
+                name=name,
+                sens_slot=i + num_batteries,
+                act_slot=i - num_sensors - 1,
+                config_entry=config_entry,
+            )
+
+            # Store the entity by ID in hass.data
+            hass.data[DOMAIN][sensor.unique_id] = sensor
+
+            sensors.append(sensor)
+
+    if sensors:
+        async_add_entities(sensors)
+
+
+class VegeHubSwitch(SwitchEntity):
```

Comment:
you can only add one platform in a single PR. Decide for either switch or sensors and remove the other

Reply:
I removed switch support. Commit 984d0d07af88b0df51a701d4da3195e98033ec50.